ams/server/src/main/java/com/netease/arctic/server/optimizing/plan/MixedHivePartitionPlan.java

```diff
@@ -184,12 +186,12 @@ protected boolean reachHiveRefreshInterval() {
     }
 
     @Override
-    public boolean fileShouldRewrite(DataFile dataFile, List<ContentFile<?>> deletes) {
+    public boolean segmentShouldRewrite(DataFile dataFile, List<ContentFile<?>> deletes) {
```

Comment:
For Mixed Hive Table, if it is not full optimizing, we can't rewrite files in hive location, even if they are fragment files.

So I suggest reverting this method name and **not** handling the fragment files and segment files separately in the `AbstractPartitionPlan`
```java
  @Override
  public boolean addFile(DataFile dataFile, List<ContentFile<?>> deletes) {
    boolean added = evaluator().addFile(dataFile, deletes);
    if (added) {
      if (evaluator().isFragmentFile(dataFile)) { // handle fragment files here 
        rewriteDataFiles.put(dataFile, deletes);
      } else if (evaluator().segmentShouldRewrite(dataFile, deletes)) { // handle segment files here 
        // Segment files whose pos delete rate is greater than the value of
        // `self-optimizing.major.trigger.duplicate-ratio`
        rewriteDataFiles.put(dataFile, deletes);
      } else if (evaluator().isUndersizedSegmentFile(dataFile)) {
        undersizedSegmentFiles.put(dataFile, deletes);
      } else if (evaluator().segmentShouldRewritePos(dataFile, deletes)) {
        rewritePosDataFiles.put(dataFile, deletes);
      } else {
        added = false;
      }
    }
    if (!added) {
      reservedDeleteFiles(deletes);
    }
    return added;
  }
```

Reply:
The method `isUndersizedSegmentFile` should also be rewritten in the `MixedHivePartitionPlan`, like 
```java
    @Override
    protected boolean isUndersizedSegmentFile(DataFile dataFile) {
      return !inHiveLocation(dataFile) && super.isUndersizedSegmentFile(dataFile);
    }
```

Reply:
The reason why we should not rewrite files in hive location is a little complicated, and it is mainly because the design specifies that data can only be synchronized from non-Hive locations to Hive locations. Once data synchronization occurs from a Hive location to a non-Hive location, there is a risk of data duplication.

Reply:
Done.